### DISCUSSION TITLE
framework not found UserNotifications

### DISCUSSION THREAD

**Discussion by xiongyejun** [fyne-io/fyne#3264] [Q&A] [answered]
go version go1.18.6 darwin/amd64
fyne/v2 v2.2.3

$ go build -buildvcs=false -v
fyne.io/fyne/v2/app
# fyne.io/fyne/v2/app
ld: framework not found UserNotifications
clang: error: linker command failed with exit code 1 (use -v to see invocation)

How can I solve this problem? thank you

**Comment by andydotxyz** [comment]
Which version of macOS are you building on?

**Reply by xiongyejun** [reply]
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G14042

$ clang --version
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.7.0
Thread model: posix

According to jacalz's method, using '-tags legacy' to the compilation command, and the compilation is successful.

thank you all thesame.

**Comment by Jacalz** [comment, marked as answer]
It might be the case that you need to build using `-tags legacy` depending on what version you are on.

**Reply by xiongyejun** [reply]
Thank you very much. 

The compilation is successful according to your method, but I don't understand the role of using -tags legacy in compiling.

Please give me more help，thank you.

**Reply by Jacalz** [reply]
The regular builds use a set of notification APIs that are not available on older versions of macOS. Using `-tags legacy` tells the compiler to instead compile the files that we have added to support the old system.

**Reply by xiongyejun** [reply]
> The regular builds use a set of notification APIs that are not available on older versions of macOS. Using `-tags legacy` tells the compiler to instead compile the files that we have added to support the old system.

thank you.
